src/series25.rs

```diff
@@ -158,3 +164,84 @@ impl<SPI: Transfer<u8>, CS: OutputPin> Flash<SPI, CS> {
         spi_result.map(|_| ()).map_err(Error::Spi)
     }
 }
+
+impl<SPI: Transfer<u8>, CS: OutputPin> BlockDevice<u32, SPI, CS> for Flash<SPI, CS> {
+    const SECTOR_LENGTH: usize = 4096;
+
+    unsafe fn erase_bytes_unchecked(
+        &mut self,
+        addr: u32,
+        amount: usize,
+    ) -> Result<(), Error<SPI, CS>> {
+        let amount = amount / Self::SECTOR_LENGTH;
+        for c in 0..amount {
+            self.write_enable()?;
+
+            let current_addr: u32 = (addr as usize + c * 256).try_into().unwrap();
+            let mut cmd_buf = [
+                Opcode::SectorErase as u8,
+                (current_addr >> 16) as u8,
+                (current_addr >> 8) as u8,
+                current_addr as u8,
+            ];
+            self.command(&mut cmd_buf)?;
+
+            let mut done = false;
+            // Wait until the erase is done
+            // TODO: maybe exchange this with a delay
+            while !done {
+                let status = self.read_status()?;
+                done = (status & Status::BUSY).is_empty();
+            }
+        }
+
+        Ok(())
+    }
+
+    fn write_bytes(&mut self, addr: u32, data: &mut [u8]) -> Result<(), Error<SPI, CS>> {
+        for (c, chunk) in data.chunks_mut(256).enumerate() {
+            self.write_enable()?;
+
+            let current_addr: u32 = (addr as usize + c * 256).try_into().unwrap();
+            let mut cmd_buf = [
+                Opcode::PageProg as u8,
+                (current_addr >> 16) as u8,
+                (current_addr >> 8) as u8,
+                current_addr as u8,
+            ];
+
+            self.cs.set_low().map_err(Error::Gpio)?;
+            let mut spi_result = self.spi.transfer(&mut cmd_buf);
+            if spi_result.is_ok() {
+                spi_result = self.spi.transfer(chunk);
+            }
+            self.cs.set_high().map_err(Error::Gpio)?;
+            spi_result.map(|_| ()).map_err(Error::Spi)?;
```

Comment:
Hmm, this chip-selection pattern could also be moved into a separate method (though you don't have to do that in this PR). Something like:

```rust
fn with_cs(&mut self, f: impl FnOnce(&Self) -> Result<...>) -> Result<...>
```